Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/HDRColorGrading.azsl

```diff
@@ -0,0 +1,212 @@
+}
+
+float3 ColorGradeKelvinColorTemp(float3 frame_color, float kTemp, float mix, float lumPreservation)
+{
+    float3 kColor    = TransformColor(KelvinToRGB(kTemp), ColorSpaceId::LinearSRGB, ColorSpaceId::ACEScg);
+    float3 oLum      = RGBToHSL(frame_color.rgb);
+    float3 blended   = lerp(frame_color.rgb, frame_color.rgb * kColor.rgb, mix);
+    float3 resHSV    = RGBToHSL(blended.rgb);
+    float3 resRGB    = HSLToRGB(float3(resHSV.xy, oLum.z));
+    return lerp(blended.rgb, resRGB.rgb, lumPreservation);
+}
+
+// pow(f, e) won't work if f is negative, or may cause inf/NAN.
+float3 NoNanPow(float3 base, float3 power)
+{
+    return pow(max(abs(base), float3(FLOAT_EPSILON, FLOAT_EPSILON, FLOAT_EPSILON)), power);
+}
+
+float3 ColorGradeSplitTone (float3 frame_color, float balance, float mix) 
+{
+    float3 frame_splitTone = NoNanPow(frame_color, 1.0 / 2.2);
+	float t = saturate(CalculateLuminance(saturate(frame_splitTone), ColorSpaceId::ACEScg) + balance);
+	float3 shadows = lerp(0.5, PassSrg::m_splitToneShadowsColor.rgb, 1.0 - t);
+	float3 highlights = lerp(0.5, PassSrg::m_splitToneHighlightsColor.rgb, t);
+	frame_splitTone = BlendMode_SoftLight(frame_splitTone, shadows);
+	frame_splitTone = BlendMode_SoftLight(frame_splitTone, highlights);
+    frame_splitTone = NoNanPow(frame_splitTone, 2.2);
+	return lerp(frame_color.rgb, frame_splitTone.rgb, mix);
+}
+
+float3 ColorGradeChannelMixer (float3 frame_color) 
+{
+	return mul(float3x3(PassSrg::m_channelMixingRed.rgb,
+            PassSrg::m_channelMixingGreen.rgb,
+            PassSrg::m_channelMixingBlue.rgb),
+		frame_color);
+}
+
+float3 ColorGradeShadowsMidtonesHighlights (float3 frame_color, float shadowsStart, float shadowsEnd, 
+                                            float highlightsStart, float highlightsEnd, float mix,
+                                            float4 shadowsColor, float4 midtonesColor, float4 highlightsColor) 
+{
+    float3 shadowsColorACEScg = TransformColor(shadowsColor.rgb, ColorSpaceId::LinearSRGB, ColorSpaceId::ACEScg);
+    float3 midtonesColorACEScg = TransformColor(midtonesColor.rgb, ColorSpaceId::LinearSRGB, ColorSpaceId::ACEScg);
+    float3 highlightsColorACEScg = TransformColor(highlightsColor.rgb, ColorSpaceId::LinearSRGB, ColorSpaceId::ACEScg);
+
+	float cLuminance = CalculateLuminance(frame_color, ColorSpaceId::ACEScg);
+	float shadowsWeight = 1.0 - smoothstep(shadowsStart, shadowsEnd, cLuminance);
+	float highlightsWeight = smoothstep(highlightsStart, highlightsEnd, cLuminance);
+	float midtonesWeight = 1.0 - shadowsWeight - highlightsWeight;
+
+    float3 frame_smh = frame_color * shadowsColorACEScg * shadowsWeight +
+	                   frame_color * midtonesColorACEScg * midtonesWeight +
+	                   frame_color * highlightsColorACEScg * highlightsWeight;
+	return lerp(frame_color.rgb, frame_smh.rgb, mix);
+}
+
+float3 ColorGrade (float3 frame_color) 
+{
+    frame_color = min(frame_color, 60.0);
```

Comment:
This is sus - why is 60 the max?

Reply:
@HogJonny-AMZN might know, I am unsure.

Reply:
We probably don't need this, I it came from the tutorial I was following and I think is probably related to a linear but non-ACES rendering path.